draft-rescorla-tls-esni.md

```diff
@@ -402,20 +396,31 @@ serves the connection directly (if in Shared Mode), in which case
 it executes the steps in the following section, or forwards
 the TLS connection to the hidden server (if in Split Mode).
 
+## Shared Mode Hidden Server Behavior
 
-## Hidden Server Behavior
+A server operating in Shared Mode uses PaddedServerNameList.sni as
+if it were the "server_name" extension to finish the handshake. It
+SHOULD pad the Certificate message with the padding(21) extension {{!RFC7685}}
+to the length of the largest possible Certificate covered by the same
+ESNI key.
```

Comment:
Why do you think we should use this rather than pad at the record layer?

Reply:
IIUC one could pad Certificate messages offline, if you wanted. One could also pad at the record later. I’ll just note one or the other.